### DISCUSSION TITLE
Apollo supported vehicle sensors!

### DISCUSSION THREAD

**Discussion by daohu527** [ApolloAuto/apollo#15214] [Ideas]
## Lidar

- Velodyne
  - VLS-128
  - VLP-32C
  - VLP-16
- Hesai
  - Pandora
- Robosense
  - Rs-LiDAR-16
 - Lslidar 
   - C16 Series

## Camera

- Leopard
  - LI-USB30-AR023ZWDR

## Radar

- Continental
  - ARS408-21
- Racobit
  - B01HC

## GNSS
docs/06_Perception/how_to_add_a_gps_receiver.md

- Asensing
  - P-Box 570
  - P-Box 571
- CHCNAV
  - CGI-610
- NovAtel
  - ProPak6
  - PwrPak 7D
 - StarNeto 
   - Newton-M1

## Computing unit

- Neousys
  - Nuvo-6108GC
  - Nuvo-8108GC
- TZTEK
  - TADC-Orin-2

**Comment by CesarLiu** [comment]
for GNSS, we use iMAR iNAT: https://github.com/fortiss/apollo/tree/dev_fortiss/modules/drivers/imar

**Reply by CesarLiu** [reply]
for radar, https://github.com/fortiss/apollo/tree/umrr_radar_driver/modules/drivers/radar/umrr_radar/python, smartmicro.

**Reply by daohu527** [reply]
Thanks! We will evaluate whether to add or release the corresponding hardware support process

**Comment by HandsomeAIccx** [comment]
Some USB_CAMERA outputs are in raw format, such as 'BA12',will it be supported in the future?

**Reply by daohu527** [reply]
The current camera format conversion is in `modules/drivers/camera/usb_cam.cc`. For now Apollo support `YUYV` and `RGB`

I would like to ask if there are scenarios for this format, or does it have special requirements for hardware? This way we can better evaluate whether to support it

**Comment by Tipriest** [comment]
will ouster lidar like ouster32 be supported in the future? if the answer is yes, may i ask about the accurate time that ouster series lidar will be supported?

**Reply by daohu527** [reply]
@Tipriest Discussions have been held here and are ongoing  https://github.com/ouster-lidar/ouster-ros/issues/287
